src/global.ts

```diff
@@ -98,6 +99,11 @@ export class Global {
    */
   public static async createDir(dirPath?: string): Promise<void> {
     dirPath = dirPath ? path.join(Global.SFDX_DIR, dirPath) : Global.SFDX_DIR;
-    await mkdirp(dirPath, '700');
+    try {
+      await fs.promises.mkdir(dirPath, { recursive: true });
+      await fs.promises.chmod(dirPath, 0o700);
```

Comment:
the permissions can be specified when creating the folder, see: https://nodejs.org/api/fs.html#fspromisesmkdirpath-options